Content.Shared/Backmen/Surgery/Body/SharedBodySystem.Integrity.cs

```diff
@@ -96,45 +115,76 @@ public override void Update(float frameTime)
         DamageSpecifier damage,
         BodyPartType targetType,
         TargetBodyPart targetPart,
         bool canSever,
+        bool evade,
         float partMultiplier)
     {
-        if (
-            partEnt.Comp.Body is not {} body ||
-            !TryComp<MobStateComponent>(body, out var mobState))
+        if (partEnt.Comp.Body is not {} body)
             return;
 
-        foreach (var (damageType, damageValue) in damage.DamageDict)
-        {
-            if (damageValue.Float() == 0
-                || TryEvadeDamage((body, mobState), GetEvadeChance(targetType)))
-                continue;
+        _proto.TryIndex<DamageGroupPrototype>("Brute", out var proto);
 
-            var modifier = GetDamageModifier(damageType);
-            var partModifier = GetPartDamageModifier(targetType);
-            var integrityDamage = damageValue.Float() * modifier * partModifier * partMultiplier;
+        if (!TryEvadeDamage(partEnt.Comp.Body.Value, GetEvadeChance(targetType)) || evade)
+        {
             TryChangeIntegrity(partEnt,
-                integrityDamage,
-                canSever && _severingDamageTypes.Contains(damageType),
+                damage * partMultiplier,
+                // This is true when damage contains at least one of the brute damage types
+                canSever && damage.TryGetDamageInGroup(proto!, out var dmg) && dmg > FixedPoint2.Zero,
                 targetPart,
-                out var severed);
-
-            if (severed)
-                break;
+                out _);
         }
     }
 
-    public void TryChangeIntegrity(Entity<BodyPartComponent> partEnt,
-        float integrity,
+    /// <summary>
+    /// Adds damage to the body part and updates things about Integrity levels.
+    /// </summary>
+    public void TryChangeIntegrity(
+        Entity<BodyPartComponent> partEnt,
+        DamageSpecifier damage,
+        bool canSever,
+        TargetBodyPart? targetPart,
+        out bool severed)
+    {
+        severed = false;
+        if (!_timing.IsFirstTimePredicted || !_queryTargeting.HasComp(partEnt.Comp.Body))
+            return;
+
+        var partIdSlot = GetParentPartAndSlotOrNull(partEnt)?.Slot;
+        var integrity = partEnt.Comp.Integrity;
+
+        partEnt.Comp.Damage.ExclusiveAdd(damage);
+        partEnt.Comp.Damage.ClampMin(partEnt.Comp.MaxIntegrity); // No over-healing!
+
+        if (canSever
+            && !HasComp<BodyPartReattachedComponent>(partEnt)
+            && !partEnt.Comp.Enabled
+            && integrity >= partEnt.Comp.SeverIntegrity
+            && partIdSlot is not null)
+            severed = true;
+
+        CheckBodyPart(partEnt, targetPart, severed);
+
+        if (severed && partIdSlot is not null)
+            DropPart(partEnt);
+
+        Dirty(partEnt, partEnt.Comp);
+    }
+
+    /// <summary>
+    /// Sets damage to the body part and updates things about Integrity levels.
+    /// Same as TryChangeIntegrity, but this can be used more fluently.
```

Comment:
A bit of a confusing description. I would say instead "Same as TryChangeIntegrity, except this one sets a given value rather than adding or subtracting"